java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -725,7 +729,19 @@ private void executeVariable(VariableTree variableTree, @Nullable Tree terminato
       } else if (variableTree.type().symbolType().is("boolean")) {
         sv = SymbolicValue.FALSE_LITERAL;
       } else if (variableTree.parent().is(Tree.Kind.CATCH)) {
-        sv = constraintManager.createSymbolicValue(variableTree);
+        Type caughtType = variableTree.symbol().type();
+        Type exceptionType = null;
+        if (programState.peekValue() instanceof SymbolicValue.ExceptionalSymbolicValue) {
+          ProgramState.Pop unstack = programState.unstackValue(1);
+          programState = unstack.state;
+          // use the Exceptional SV from the stack
+          sv = unstack.values.get(0);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)